mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSNode.kt

```diff
@@ -129,9 +131,15 @@ data class MPSNode(val node: SNode) : IDeprecatedNodeDefaults {
     }
 
     override fun setReferenceTarget(link: IReferenceLink, target: INode?) {
-        val ref = node.references.first { MPSReferenceLink(it.link).getUID() == link.getUID() }
+        val refLink = when (link) {
+            is MPSReferenceLink -> link.link
+            else -> node.references.find { MPSReferenceLink(it.link).getUID() == link.getUID() }?.link
+                ?: node.concept.referenceLinks.find { MPSReferenceLink(it).getUID() == link.getUID() }
+                ?: SReferenceLinkAdapterById(SReferenceLinkId.deserialize(link.getUID()), "")
```

Comment:
A similar change is required in `setProperty` and `addNewChild`.

Reply:
Also for `moveChild`.

Reply:
Done